src/js/actions/libraries.js

```diff
@@ -793,28 +794,35 @@ define(function (require, exports) {
                         // So, although the command does not resolve until the placement has finished,
                         // it does resolve before the layer model has been updated because the event
                         // has not yet been received. For more details see issue #2177.
-                        return descriptor.playObject(placeObj, { synchronous: false })
+                        var placeElementPromise = descriptor.playObject(placeObj, { synchronous: false })
                             .bind(this)
                             .finally(function () {
                                 return this.transfer(policyActions.removeKeyboardPolicies, policyID, true);
                             });
+                            
+                        // Dropping graphic asset with ALT/OPT modifier will expand the asset (instead of adding 
+                        // CLSO layer) and result in creating multiple layers in Photoshop. Unlike regualr drop 
+                        // event (without modifier), this will not trigger the "placeEvent" handler, so we need 
+                        // to get the new layer IDs through the difference between the previous layer IDs and the
+                        // next layer IDs.
+                        //
+                        // FIXME: we should instead get IDs back from Photoshop when layers are placed with modifier, 
+                        //        so we don't have to get all the layer IDs.
+                        if (hasAlt) {
+                            placeElementPromise.then(function () {
```

Comment:
Not a fan of the structure here. If alt is being pressed, you wait for that promise to fdinish, and then update the document, but you return the placeElement promise already.

What you can do instead is: 

`placeElementPromise = placeElementPromise.then(function ...`

and `return placeElementPromise` as usual at the end.